src/ctelnet.cpp

```diff
@@ -1994,9 +2001,9 @@ void cTelnet::setGMCPVariables(const QByteArray& msg)
         mpHost->updateProxySettings(mpDownloader);
         auto request = QNetworkRequest(QUrl(url));
         mudlet::self()->setNetworkRequestDefaults(url, request);
-        QNetworkReply* reply = mpDownloader->get(request);
+        packageDownloadReply = mpDownloader->get(request);
         mpProgressDialog = new QProgressDialog(tr("downloading game GUI from server"), tr("Cancel", "Cancel download of GUI package from Server"), 0, 4000000, mpHost->mpConsole);
-        connect(reply, &QNetworkReply::downloadProgress, this, &cTelnet::setDownloadProgress);
+        connect(packageDownloadReply, &QNetworkReply::downloadProgress, this, &cTelnet::setDownloadProgress);
```

Comment:
Like before - you need to wire up the cancel button to the SLOT that handles it.